backend/dataall/core/environment/cdk/environment_stack.py

```diff
@@ -138,6 +138,9 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
         self.environment_admins_group: EnvironmentGroup = self.get_environment_admins_group(
             self.engine, self._environment
         )
+        # Create test role for integration tests
+        if os.getenv('INTEGRATION_TESTS', None) == 'True':
```

Comment:
You can infer this by checking if the `/{resource_prefix/{env_name}/testdata` SSM param is populated

Reply:
or of working locally check for the testdata file right?

Reply:
For the same deployment with autoapproval enabled we will deploy "test environments" as part of the testing and "real environments". The SSM param will exist no matter what no?

Reply:
Summarizing, we would recommend users to:
- have a dev environment with approval_tests enabled - all environments in the actual deployment and in the codebuild  test session include an integration test role (very permissive role)
- have a prod environment with approval_tests disabled 

My concern is that users enable approval_tests in prod and start deploying a very open IAM role in all environments. I would only deploy the test role when we are doing tests (if possible)

Reply:
The more I think about it the more I agree with you. Having an environment variable makes it quite complex, because we need to pass the env variable from the testing environment to API to ECS. I am good with having guardrails as you suggested and clear warnings and recommendations

Reply:
There is a problem with checking in SSM and that is that the SSM parameter is stored in the tooling account right? the ECS task that deploys the environment is in the infra account, so it would need to read the env from the tooling account. We might need to create a different SSM parameter in the infra account

Reply:
Recapping alternatives:
- Env variables - very complex because we are passing them from env to env 
- SSM - very complex because of the accounts isolation, plus we cannot distinguish which environments are test environments and which ones are prod
- parameter in API call - seems to be the best option, we can differentiate between test and no-test environments and we pass the info directly to the API call

Reply:
Solved and implemented in PR #1379